### ISSUE TITLE
# Aborted (core dumped) in `flow.nn.Conv2d`

### ISSUE THREAD

**Issue by deepliao** [Oneflow-Inc/oneflow#10638]
## Summary

When applying `flow.nn.Conv2d` to an input whose spatial dimensions are too small for the given kernel size, stride, and padding, OneFlow attempts to compute a convolution with a resulting output shape of `(N, C_out, 0, 0)`. This leads to a crash with **"Aborted (core dumped)"**, rather than raising a proper error indicating invalid output dimensions. The root cause is the lack of shape validation in the convolution operator.

## Code to reproduce bug

```python
import oneflow as flow
import numpy as np

np.random.seed(42)
flow.manual_seed(42)

conv = flow.nn.Conv2d(
    in_channels=3,
    out_channels=64,
    kernel_size=(5, 5),
    stride=(1, 1),
    padding=(0, 0),
    dilation=(1, 1),
    groups=1,
    bias=True
)

# Input spatial size (4, 4) is too small for 5x5 kernel without padding
input = flow.tensor(
    np.random.randn(2, 3, 4, 4),
    dtype=flow.float32
)

# Forward pass — leads to (2, 64, 0, 0) output and backend crash
output = conv(input)
print("Output shape:", output.shape)

```

output:
```
*** Check failure stack trace: ***
    @     0x7f90dad089ca  google::LogMessage::Fail()
    @     0x7f90dad08cb2  google::LogMessage::SendToLog()
    @     0x7f90dad08537  google::LogMessage::Flush()
    @     0x7f90dad0b0a9  google::LogMessageFatal::~LogMessageFatal()
    @     0x7f90d0aeb188  oneflow::ep::primitive::broadcast_matmul::Simplify()
    @     0x7f90d0ae8583  oneflow::ep::primitive::broadcast_matmul::internal::(anonymous namespace)::BroadcastMatmulImpl<>::Launch()
    @     0x7f90d0730d2d  oneflow::ep::primitive::(anonymous namespace)::BatchMatmulImpl::Launch()
    @     0x7f90d07317e6  oneflow::ep::primitive::(anonymous namespace)::MatmulImpl::Launch()
    @     0x7f90d50eb997  oneflow::(anonymous namespace)::ConvCpuKernel<>::Compute()
    @     0x7f90d6a86536  oneflow::one::StatefulOpKernel::Compute()
    @     0x7f90d4d20cab  oneflow::vm::OpCallInstructionUtil::Compute()
    @     0x7f90d4d1e787  oneflow::vm::OpCallInstructionPolicy::Compute()
    @     0x7f90d4d1a5bc  oneflow::vm::Instruction::Compute()
    @     0x7f90d4d18a6f  oneflow::vm::EpStreamPolicyBase::Run()
    @     0x7f90d4d24086  oneflow::vm::StreamPolicy::RunIf()
    @     0x7f90d4d2b6de  oneflow::vm::ThreadCtx::TryReceiveAndRun()
    @     0x7f90d4d2dd18  oneflow::(anonymous namespace)::WorkerLoop()
    @     0x7f90d4d2e11f  _ZNSt6thread11_State_implINS_8_InvokerISt5tupleIJPFvPN7oneflow2vm9ThreadCtxERKSt8functionIFvS6_EEES6_ZNS3_14VirtualMachine15CreateThreadCtxENS3_6SymbolINS3_6DeviceEEENS3_10StreamTypeEmEUlS6_E3_EEEEE6_M_runEv
    @     0x7f90dad1d40f  execute_native_thread_routine
    @     0x7f91b3b85609  start_thread
    @     0x7f91b3aac293  clone
Aborted (core dumped)
```

##  System Information
OneFlow installation: pip
OS: Ubuntu 20.04
OneFlow version (run python3 -m oneflow --doctor):
```
path: ['/root/miniconda3/envs/myconda/lib/python3.8/site-packages/oneflow']
version: 0.9.0
git_commit: 381b12c
cmake_build_type: Release
rdma: True
mlir: True
```
Python version: 3.8
CUDA driver version: 11.6
GPU models: NVIDIA A16
